Physical_oM/Materials/Options/DensityExtractionOptions.cs

```diff
@@ -0,0 +1,79 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2022, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Base;
+using BH.oM.Quantities.Attributes;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+
+namespace BH.oM.Physical.Materials.Options
+{
+    [Description("Settings for extraction of Density from a Physical Material.")]
+    public class DensityExtractionOptions : IObject
+    {
+        /***************************************************/
+        /**** Properties                                ****/
+        /***************************************************/
+
+        [Description("A specific type of IDensityProvider to limit the search to. If null all IDensityProviders on the material are considered.")]
+        public virtual Type Type { get; set; } = null;
+
+        [Description("If true, and material does not contain a IDensityProvider of the specified type, DensityExtraction falls back to look for other densities. If false and material does not contain the type DensityExtraction exits.")]
+        public virtual bool AllowFallbackIfNoType { get; set; } = false;
+
+        public virtual DensityExtractionType ExtractionType { get; set; } = DensityExtractionType.AllMatching;
+
+        [Ratio]
+        [Description("The ratio tolerance for considering the value of the densities as equal. Comparison is made by comparing the differance between min and max over their mean.")]
```

Comment:
It could be that I'm just super tired right now, but `Comparison is made by comparing the difference between min and max over their mean.` isn't making a lot of sense. Can you give an example of this calculation?

```suggestion
        [Description("The ratio tolerance for considering the value of the densities as equal. Comparison is made by comparing the difference between min and max over their mean.")]
```

Reply:
Will have a proper read-through myself. Just copy-pasted the text from a method :P